### PR TITLE
Added new metric to monitor CPU throttling

### PR DESCRIPTION
**Pull request by ggaurav10**
**What this PR does / why we need it**:
Added new metric to monitor CPU throttling

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Added new metric to monitor CPU throttling
```

### CODE REVIEW

**Comment by vpnachev** [comment]
@ggaurav10 could you explain why these metrics are needed?
My concern is to not overload prometheus performance.

**Review by rfranzke — DISMISSED** [review]

**Comment by ggaurav10** [comment]
yes @vpnachev, this was my concern as well, hence the delay between making the change and raising a PR :)
Recent issue in our seeds, and a possibility of CPU throttling was the motivation behind finally raising this PR. This would help in figuring out if any of the pods are/were cpu throttled. I am not sure if there is any other way this could be figured out, since unlike memory (oomkills) there is no other trace of cpu throttling.

**Review by amshuman-kr — CHANGES_REQUESTED** [review]
@ggaurav10 Thanks for adding the metric.

Can you please also add seed-wide aggregate metrics (sum, sum_by_container and sum_cp) [here](https://github.com/gardener/gardener/blob/master/charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml)?

LGTM otherwise.

File: charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml [hunk 1]
```diff
@@ -5,6 +5,12 @@ groups:
   - record: seed:container_cpu_usage_seconds_total:sum_by_container
     expr: sum(rate(container_cpu_usage_seconds_total[5m])) by (container)
 
+  - record: seed:container_cpu_cfs_throttled_seconds_total:sum_by_container
+    expr: sum(rate(container_cpu_cfs_throttled_seconds_total[5m])) by (container)
```
Comment (amshuman-kr) [inline charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml:9]: Does is make sense to aggregate the rate? `container_cpu_cfs_throttled_seconds_total` is not about how much CPU cores are throttled, right?

File: charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml [hunk 2]
```diff
@@ -5,6 +5,12 @@ groups:
...
+
+  - record: seed:container_cpu_cfs_throttled_periods_total:sum_by_container
+    expr: sum(rate(container_cpu_cfs_throttled_periods_total[5m])) by (container)
```
Comment (amshuman-kr) [inline charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml:12]: Same as above.

**Review by amshuman-kr — COMMENTED** [review]
Is the aggregation based on rate appropriate?

**Review by amshuman-kr — APPROVED** [review]
Thanks @ggaurav10 for the links.

https://rancher.com/docs/rancher/v2.x/en/cluster-admin/tools/monitoring/expression/
https://opensource.zalando.com/skipper/operation/operation/

**Review by wyb1 — APPROVED** [review]
/lgtm

**Review by vpnachev — APPROVED** [review]
/approve

**Comment by mwennrich** [comment]
So this will be in the next release?

**Comment by rfranzke** [comment]
@mwennrich yes, planned for this week.

**Pull request merged**